packages/terra-table/src/Table.jsx

```diff
@@ -1,68 +1,401 @@
+    primaryData = { label: rowData.discloseAction.discloseLabel, isCurrent: rowData.discloseAction.isDisclosed };
+  } else if (rowStyle === 'toggle' && rowData.toggleAction) {
+    rowMetaData = rowData.toggleAction.metaData;
+    rowOnAction = rowData.toggleAction.onToggle;
+    rowActiveState = rowData.toggleAction.isToggled;
+  }
+
+  return (
+    <Row
+      {...rowData.attrs}
+      key={rowData.key}
+      aria-rowindex={rowData.index || rowIndex}
+      metaData={rowMetaData}
+      isSelectable={rowStyle === 'toggle' || rowStyle === 'disclose' || checkStyle === 'toggle'}
+      isSelected={rowActiveState}
+      onSelect={rowOnAction}
+      isDisabled={rowData.isDisabled}
+      isStriped={rowData.isStriped}
+      dividerStyle={dividerStyle}
+    >
+      {createCheckCell(rowData, rowStyle, checkStyle)}
+      {rowData.cells.map((cell, colIndex) => {
+        const columnId = headerData && headerData.cells ? headerData.cells[colIndex].id : undefined;
+        const columnWidth = columnWidths ? columnWidths[colIndex] : undefined;
+        const discloseData = colIndex === primaryIndex ? primaryData : undefined;
+        return createCell(cell, sectionId, columnId, columnWidth, discloseData);
+      })}
+      {createChevronCell(rowStyle, hasChevrons)}
+    </Row>
+  );
+};
+
+const createSections = (headerIndex, sectionData, headerData, columnWidths, rowStyle, checkStyle, hasChevrons, dividerStyle) => {
+  if (!sectionData) {
+    return { sections: undefined, sectionIndex: headerIndex };
+  }
+
+  let rowIndex = headerIndex;
+  const sections = sectionData.map((section) => {
+    if (section.sectionHeader) {
+      const header = section.sectionHeader;
+      rowIndex += 1;
+      return (
+        <Section
+          id={header.id}
+          key={header.key}
+          aria-rowindex={header.index || rowIndex}
+          title={header.title}
+          isCollapsed={header.isCollapsed}
+          isCollapsible={!!header.onToggle}
+          metaData={header.metaData}
+          numberOfColumns={header.numberOfColumns}
+          onSelect={header.onToggle}
+        >
+          {section.rows ? section.rows.map(rowData => {
+            rowIndex += 1;
+            return createRow(rowData, rowIndex, header.id, headerData, columnWidths, rowStyle, checkStyle, hasChevrons, dividerStyle);
+          }) : undefined}
+        </Section>
+      );
+    }
+    if (section.rows) {
+      return section.rows.map(rowData => {
+        rowIndex += 1;
+        return createRow(rowData, rowIndex, null, headerData, columnWidths, rowStyle, checkStyle, hasChevrons, dividerStyle);
+      });
+    }
+    return undefined;
+  });
+
+  return { sections, sectionIndex: rowIndex };
+};
+
+const createHeader = (headerData, columnWidths, rowStyle, checkStyle, hasChevrons) => {
+  if (!headerData || !headerData.cells) {
+    return { headerIndex: 0, header: undefined };
+  }
+
+  return {
+    headerIndex: 1,
```

Comment:
Could we return the numberOfColumns here to use to create the sections?

```
 return {
    headerIndex: 1,
    numberOfColumns: headerData.length,
    header: ( ...
```
then
```diff
const unpackTableData = (headerData, sectionData, columnWidths, rowStyle, checkStyle, hasChevrons, dividerStyle) => {
-  const { headerIndex, header } = createHeader(headerData, columnWidths, rowStyle, checkStyle, hasChevrons);
+  const { headerIndex, numberOfColumns, header } = createHeader(headerData, columnWidths, rowStyle, checkStyle, hasChevrons);
-  const { sectionIndex, sections } = createSections(headerIndex, sectionData, headerData, columnWidths, rowStyle, checkStyle, hasChevrons, dividerStyle);
+  const { sectionIndex, sections } = createSections(headerIndex, sectionData, headerData, columnWidths, rowStyle, checkStyle, hasChevrons, dividerStyle, numberOfColumns);
  return { rowCount: sectionIndex, header, sections };
};

```
&&&

```diff
- const createSections = (headerIndex, sectionData, headerData, columnWidths, rowStyle, checkStyle, hasChevrons, dividerStyle) => {
+ const createSections = (headerIndex, sectionData, headerData, columnWidths, rowStyle, checkStyle, hasChevrons, dividerStyle, numberOfColumns) => {
   ......
      return (
        <Section
           ...
-          numberOfColumns={header.numberOfColumns}
+          numberOfColumns={numberOfColumns}